keps/sig-instrumentation/1748-pod-resource-metrics/README.md

```diff
@@ -0,0 +1,425 @@
+
+Future features may alter this model, such as the pod resizing proposal, and those rules will be automatically reflected in the resources handled by the system.
+
+
+### User Stories (optional)
+
+#### Story 1
+
+As an administrator of a Kubernetes cluster, I can easily see the requested resources for the pods on the cluster and compare those to the actual usage of those pods with our instrumentation pipeline.
+
+#### Story 2
+
+As an extender of Kubernetes, my extended resources used by the scheduler should be easily queryable by an administrator without requiring Kubernetes code changes or adding new components to the system.
+
+### Story 3
+
+An administrator of the Kubernetes cluster should be able to see an aggregate representation of actual resource consumption metrics that is consistent with the resource request and scheduling model in order to measure resources that are imprecisely sized.
+
+### Notes/Constraints/Caveats (optional)
+
+The `kube-state-metrics` component of Kubernetes already exposes a lower fidelity model of resource requests from pods. However, because it is focused on exposing the attributes of the pod resource model rather than their calculated meaning, it captures an imprecise representation of the metrics Kubelet and the scheduler use to make decisions. This representation is not suitable for making complete decisions on *why* a resource is not schedulable, or to accurately capture how the scheduler views the resource model. In order to practically model these resources, a number of high cardinality metrics would need to be added and a fairly complicated calculation would have to be performed in the metrics aggregator that would subtly differ from the decisions made by the Kubelet or scheduler. Instead we recommend closely matching the implementation the scheduler uses to make decisions (documented and part of our API, but not explicitly modelled outside of code) to represent the key metrics for the resource model at significantly reduced overall cardinality. For this reason in kube-state-metrics the resource metrics would be deprecated and removed at the point where this proposal goes GA.
```

Comment:
>  For this reason in kube-state-metrics the resource metrics would be deprecated and removed at the point where this proposal goes GA.

I am not sure this should be a hard cut when this goes GA, or even ever. As these metrics are opt-in that means for some kube-state-metrics resource metrics would be only metrics they can scrape.

Also kube-state-metrics is a standalone optional exporter not tied to strictly one version of Kubernetes or its releases, meaning if these go GA in 1.21 for example, some folks can use kube-state-metrics at that point still in previous versions.

I would say lets change the wording to something along the lines of:
> For this reason in kube-state-metrics the resource metrics might be deprecated and removed at some point in the future, but not tied to this implementation.

And can you open a proposal for this in kube-state-metrics when this is merged, we can always amend it afterward.

Reply:
Ok.